resolwe_bio/tests/generators/test_management_commands.py

```diff
@@ -20,37 +63,46 @@ def test_generate_samples(self):
             self.assertEqual(sample_data[0].slug, 'gs-reads')
             self.assertEqual(sample_data[1].slug, 'mapping')
             self.assertEqual(sample_data[2].slug, 'expression')
+
             expr = sample_data[2]
-            # NOTE: Python 2 and 3 produce different results even when setting random.seed() to the same number
-            if six.PY2:
-                self.assertJSON(expr, expr.output['exp_json'], '', 'expressions-py2.json.gz')
-                self.assertFile(expr, 'exp', 'expressions-py2.tab.gz', compression='gzip')
-            else:
-                self.assertJSON(expr, expr.output['exp_json'], '', 'expressions-py3.json.gz')
-                self.assertFile(expr, 'exp', 'expressions-py3.tab.gz', compression='gzip')
+            # NOTE: The random.gammavariate() that is used for generating expression data produces
+            # slightly differently rounded numbers on different platforms (e.g. Linux and MacOS) so
+            # the check ignores the generated numbers
+            with mock.patch.object(self, 'assertDictEqual', self.assertDictEqualIgnoreNumbers):
```

Comment:
It checks everything else in the generated JSON, the dict structure, lists, strings, ...

The problem is that the check is "static", i.e. it compares the structure to the saved JSON file. What we want in the future is a "dynamic" check that compares the generator's output to the output of a real processor it is simulating.